aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -571,6 +571,100 @@ Tensor& cholesky_out(Tensor &result, const Tensor &self, bool upper) {
   return result;
 }
 
+template<typename scalar_t>
+static void  apply_cholesky_mod(const Tensor& self, Tensor & out, Tensor & err, bool upper, std::vector<int64_t>& infos) {
+#ifndef USE_LAPACK
+  AT_ERROR("cholesky: LAPACK library not found in compilation");
+#else
+  char uplo = upper ? 'U' : 'L';
+
+  Tensor outTag =  out;
+  Tensor selfTag = self;
+
+  double* err_data = err.data_ptr<double>();
+  auto self_matrix_stride = matrixStride(self);
+  auto batch_size = batchCount(self);
+  auto n = self.size(-2);
+  auto m = self.size(-1);
+
+  if (batch_size > 1) {
+    outTag =  out.flatten(0, -3);
+    selfTag = self.flatten(0, -3);
+  }
+  
+  scalar_t* self_data = selfTag.data_ptr<scalar_t>();
+  scalar_t* out_data = outTag.data_ptr<scalar_t>();
+  /* TODO: do this in parallel */
+  for (int64_t i = 0; i < batch_size; i++) {
+    scalar_t* out_working_ptr = &out_data[i * self_matrix_stride];
+    scalar_t* self_working_ptr = &self_data[i * self_matrix_stride];
+    double err_val = 1e-8;
+    int info;
+    lapackCholesky<scalar_t>(uplo, n, out_working_ptr, n, &info);
+    for (int64_t j=0; j<8; j++)
+    {
+      if (info == 0) {
+        break;
```

Comment:
Just to make sure: This circumvents the exceedingly slow error handling that is pointed out in #34272, right?

Reply:
As long as the jitter addition succeeds (after maximum 6 iterations) we avoid the error handling. Otherwise it still raises. I should add a corresponding test

Reply:
Raises you mean with the internal Torch checks?

Reply:
Yes, with the checks of `info` beginning on line 646